book/stages/README.md

```diff
@@ -0,0 +1,162 @@
+# Stages
+
+The `stages` lib plays a central role in syncing the node, maintaining state, updating the database and more. The stages involved in the Reth pipeline are the `HeaderStage`, `BodyStage`, `SendersStage`, and `ExecutionStage` (note that this list is non-exhaustive, and more pipeline stages will be added in the near future). Each of these stages are queued up and stored within the Reth pipeline.
+
+[Filename: crates/stages/src/pipeline.rs](https://github.com/paradigmxyz/reth/blob/main/crates/stages/src/pipeline.rs#L76)
+```rust
+pub struct Pipeline<DB: Database> {
+    stages: Vec<QueuedStage<DB>>,
+    max_block: Option<BlockNumber>,
+    events_sender: MaybeSender<PipelineEvent>,
+}
+```
+
+When the node is first started, a new `Pipeline` is initialized and all of the stages are added into `Pipeline.stages`. Then, the `Pipeline::run` function is called, which starts the pipeline, executing all of the stages continuously in an infinite loop. This process syncs the chain, keeping everything up to date with the chain tip. 
+
+Each stage within the pipeline implements the `Stage` trait which provides function interfaces to get the stage id, execute the stage and unwind the state if there was an issue during the stage execution. 
```

Comment:
```suggestion
Each stage within the pipeline implements the `Stage` trait which provides function interfaces to get the stage id, execute the stage and unwind the stage if there was an issue during the stage execution. 
```
nit: do you mean to say "state" here? not all stages touch EVM "state" so this may be a slightly confusing term

Reply:
What I meant by this is `unwind the changes to the database`. I can update the language to reflect this or reword it to whatever makes the most sense.